src/cascade/saver/db/save_model_results_to_db.py

```diff
@@ -19,33 +23,77 @@ def __getattr__(self, name):
 
 DRAWS_INPUT_FILE_PATTERN = "all_draws.h5"
 
-INTEGRAND_ID_TO_MEASURE_ID_DF = pd.DataFrame([
-    [0, 41],
-    [1, 7],
-    [2, 9],
-    [3, 16],
-    [4, 13],
-    [5, 39],
-    [6, 40],
-    [7, 5],
-    [8, 6],
-    [9, 15],
-    [10, 14],
-    [11, 12],
-    [12, 11]], columns=["integrand_id", "measure_id"])
+INTEGRAND_ID_TO_MEASURE_ID_DF = pd.DataFrame(
+    [
+        [0, 41],  # Susceptible incidence
+        [1, 7],  # Remission
+        [2, 9],  # Excess mortality rate
+        [3, 16],  # Other cause mortality rate
+        [4, 13],  # With-condition mortality rate
+        [5, 39],  # Susceptible population fraction
+        [6, 40],  # With Condition population fraction
+        [7, 5],  # Prevalence
+        [8, 42],  # Total Incidence
+        [9, 15],  # Cause-specific mortality rate
+        [10, 14],  # All-cause mortality rate
+        [11, 12],  # Standardized mortality ratio
+        [12, 11],  # Relative risk
+    ],
+    columns=["integrand_id", "measure_id"],
+)
 
 CODELOG = logging.getLogger(__name__)
 MATHLOG = logging.getLogger(__name__)
 
 
-def _normalize_draws_df(draws_df):
+def _normalize_draws_df(draws_df, execution_context):
     """The database stores measure_id rather than integrand_id."""
 
-    draws = draws_df.merge(
-        INTEGRAND_ID_TO_MEASURE_ID_DF, how="left", on="integrand_id").drop(
-        columns=["integrand_id"])
+    draws = draws_df.merge(INTEGRAND_ID_TO_MEASURE_ID_DF, how="left", on="integrand_id").drop(columns=["integrand_id"])
+
+    if not np.allclose(draws.time_lower, draws.time_upper):
+        raise ValueError(
+            "There are integrands over time intervals but we only " "know how to upload integrands for a point in time."
+        )
+
+    expected_years = sorted(get_years(execution_context))
+    actual_years = sorted(unique_floats(draws.time_lower))
+    if not np.allclose(expected_years, actual_years):
+        raise ValueError("There are times in the avgint table that don't match GBD years")
+
+    year_ids = pd.DataFrame({"year_id": expected_years, "year_float": np.array(expected_years, dtype=np.float64)})
+    draws = pd.merge_asof(draws.sort_values("time_lower"), year_ids, left_on="time_lower", right_on="year_float").drop(
+        ["time_lower", "time_upper", "year_float"], "columns"
+    )
+
+    age_groups = get_age_groups(execution_context)
+    with_age_groups = pd.merge_asof(
+        draws.sort_values("age_lower"), age_groups, left_on="age_lower", right_on="age_group_years_start"
+    )
+
+    merge_is_good = np.allclose(with_age_groups.age_lower, with_age_groups.age_group_years_start)
+    merge_is_good = merge_is_good and np.allclose(with_age_groups.age_upper, with_age_groups.age_group_years_end)
+    merge_is_good = merge_is_good and len(draws) == len(with_age_groups)
```

Comment:
I've seen this pattern in Code Complete as: `uses_point_ages = ...` `uses_point_times = ...` `draws_equal_ages` and then `if uses_point ages and uses_point_times and draws_equal_ages`.